main.py

```diff
@@ -1,58 +1,67 @@
 import requests
 import jmespath
 from bs4 import BeautifulSoup
-import re
 
-final_results = []
-titles = []
-posters = []
-rates = []
-genres = []
-directors = []
-results = {}
+HTTP_SUCCES = 200
+MOVIES_URL = 'https://www.filmweb.pl/serials/search?endRate=10&orderBy=popularity&descending=true&startRate=9{}'
 
-response = requests.get('https://www.filmweb.pl/serials/search?endRate=10&orderBy=popularity&descending=true&startRate=9')
-soup = BeautifulSoup(response.content, 'html.parser')
-results_list = soup.find(class_="resultsList hits")
+try:
+    n = int(input("How many pages do you want to scrape?"))
+except ValueError:
+    raise Exception("Invalid value")
 
-results_list_posters = results_list.find_all(class_='poster__image')
-for poster in results_list_posters:
-    posters.append(jmespath.search('content', poster))
 
-results_list_titles = results_list.find_all(class_='filmPreview__title')
-for title in results_list_titles:
-    titles.append(title.get_text())
+def find_movies_of_n_pages(n=n):
 
-results_list_rates = results_list.find_all(class_="rateBox__rate")
-for rate in results_list_rates:
-    rates.append(rate.get_text())
+    def find_movies(page):
+        response = requests.get(MOVIES_URL.format(f'&page={page}'))
+        if response.status_code == HTTP_SUCCES:
+            soup = BeautifulSoup(response.content, 'html.parser')
+            films = soup.find(class_="resultsList hits")
+            if films:
+                return films
+            else:
+                raise Exception("No such an element")
+        else:
+            raise Exception("Invalid response recived")
```

Comment:
Already better :) But I would still prefer to see a few retry's done when an Invalid Response is received, and the whole thing can be simplified:
```
if not films:
    raise Exception("Invalid page received!")

return films
```